### DISCUSSION TITLE
How to change the color of models?

### DISCUSSION THREAD

**Discussion by tueboesen** [mlflow/mlflow#8149] [Q&A]
I just started using mlflow a few weeks ago, so there is probably a simple fix to this problem.
I have been running a bunch of experiments and probably deleted some as well. 
The end result is that I have ended up with a bunch of models and some of them are really similar in their color when doing comparisons.

Is there a way to manually change the colors of some of these models?
Here is what my current colors look like:

![image](https://user-images.githubusercontent.com/22527117/229364401-168fac37-ba81-4f2f-b140-67f0b10fe227.png)

**Comment by georgeseifada** [comment]
Also having the same issue.

**Reply by scallon** [reply]
Idem

**Reply by RobertoLorusso** [reply]
I've proposed a solution in the comment below, hope it'll be helpful :)

**Comment by RobertoLorusso** [comment]
Hi, i was facing the same issue. I implemented a workaround to create a new run and log the metrics from a previous one. This is the python code. Ensure that you are tracking the new run under the same experiment of the old one and end previous runs with

```python 
import mlflow
mlflow.end_run()
mlflow.set_experiment('experiment-name-of-old-run')
```

Then you can proceed with the following instructions:

``` python

import mlflow
from mlflow.tracking import MlflowClient

previous_run_id = "your-previous-run-id"

# Initialize the MLflow client
client = MlflowClient()

# Fetch metric histories
metric_histories = {}
for metric in client.get_run(previous_run_id).data.metrics:
    metric_histories[metric] = client.get_metric_history(previous_run_id, metric)

# Start a new run
# Ensure you ended previous runs with mlflow.end_run()
with mlflow.start_run():
    # Log historical metric values
    for metric_name, metric_history in metric_histories.items():
        for metric_entry in metric_history:
            mlflow.log_metric(metric_name, metric_entry.value, step=metric_entry.step)
            print(f"{metric_name}, {metric_entry.value}, {metric_entry.step}")

```

These are the charts before and after the creation of a new run from a previous one. I hope it will be helpful :)

<img width="467" alt="Screenshot 2024-02-08 alle 13 57 53" src="https://github.com/mlflow/mlflow/assets/51060598/8f5828fb-a8d0-4ca5-95b3-2621b8e3ca08">

<img width="320" alt="Screenshot 2024-02-08 alle 14 00 27" src="https://github.com/mlflow/mlflow/assets/51060598/0e0032c0-9d47-463c-91fb-0f5c108aa8ce">

**Comment by ParamsRaman** [comment]
+1 this is a very useful feature to have in mlflow. Currently comparing and visualizing more than 2 or 3 plots is a huge pain given the subtle color palette used. Would be good to allow "line styles" and "color" configurability in the comparion plot's "configure" panel.

**Comment by gui-miotto** [comment]
This is by far my most desired feature in MLFlow: to be able to change the color of the runs.
I would be really cool to click on the color bubble of a run and select a different color. 
Sometimes I have to compare two runs across dozens of plots and they are both plotted with the "exact" same green.
It's such a pain. :-(

**Comment by ibaris** [comment]
I have the same issue here: 
![image](https://github.com/mlflow/mlflow/assets/15630368/5f11d9c3-8018-47a4-8788-c6086f60a6cf)

The color must saved somewhere in the logs right? Can't we just manually change it there?

**Comment by Engineero** [comment]
Seeing the same issue still. This would be a nice feature to have added.

<img width="343" height="338" alt="Screenshot 2025-11-04 at 13 52 08" src="https://github.com/user-attachments/assets/c0206fe9-693a-4a8c-a414-435aa57cd3be" />
